boxes.cabal

```diff
@@ -11,24 +11,29 @@ extra-source-files:  CHANGES, README.md, include/boxes.h
 author:              Brent Yorgey
 maintainer:          David Feuer <David.Feuer@gmail.com>
 build-type:          Simple
-cabal-version:       >= 1.9.2
--- Minimum Cabal version supporting test suites
+cabal-version:       >= 1.10
 
 library
   build-depends:     base >= 3 && < 5, split >=0.2 && <0.3
-  exposed-modules:   Text.PrettyPrint.Boxes
-  extensions:        CPP
+  hs-source-dirs:    lib
   include-dirs:      include
+  exposed-modules:   Text.PrettyPrint.Boxes
+                   , Text.PrettyPrint.Boxes.Internal
+                   , Text.PrettyPrint.Boxes.Imports
+  default-language:  Haskell2010
+  default-extensions: NoImplicitPrelude
+  other-extensions:  CPP
 
 Test-Suite test-boxes
   type:              exitcode-stdio-1.0
-  main-is:           Text/PrettyPrint/Tests.hs
-  build-depends:     base >= 3 && < 5, split >=0.2 && <0.3, QuickCheck
-  extensions:        CPP
+  build-depends:     QuickCheck, boxes
+  main-is:           Tests.hs
+  hs-source-dirs:    test
   include-dirs:      include
-  cpp-options:       -DTESTING
-  other-modules:     Text.PrettyPrint.Boxes
--- Export some internals so the tests can get at them
+  other-modules:     Definitions
+  default-language:  Haskell2010
+  default-extensions: NoImplicitPrelude
+  other-extensions:  CPP
```

Comment:
Wait, WHAT? No. We don't want `NoImplicitPrelude` *anywhere*, let alone *everywhere*.

Reply:
That's news to me.  Why?